packages/components/addon/components/hds/dialog/index.js

```diff
@@ -0,0 +1,101 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { assert } from '@ember/debug';
+import { getElementId } from '../form/utils/getElementId';
+
+export const DEFAULT_SIZE = 'regular';
+export const DEFAULT_TYPE = 'neutral';
+export const SIZES = ['small', 'regular', 'large'];
+export const TYPES = ['neutral', 'warning', 'critical'];
+
+export default class HdsDialogIndexComponent extends Component {
+  @action
+  setupDialog(element) {
+    this.element = element;
+    element.showModal();
+  }
+
+  @action
+  onClick(event) {
+    // This may seem counterintuitive, but the <dialog> element is the target only when the backdrop is clicked
```

Comment:
probably another point for having the backdrop as a standalone element, instead of using a pseudo-element?

Reply:
I think the suggestion makes sense from a pragmatic point of view – I have updates this to use an element (that we can call overlay, for consistency with what we have in Figma) instead of the pseudo-element. This way we can remove this logic and leverage the code we require for the `focus-trap` anyway.